src/Compilers/CSharp/Portable/Binder/Binder_Query.cs

```diff
@@ -648,7 +659,10 @@ private UnboundLambda MakeQueryUnboundLambda(RangeVariableMap qvm, ImmutableArra
         {
             return MakeQueryUnboundLambda(expression, new QueryUnboundLambdaState(this, qvm, parameters, (LambdaSymbol lambdaSymbol, Binder lambdaBodyBinder, DiagnosticBag diagnostics) =>
             {
-                return lambdaBodyBinder.BindLambdaExpressionAsBlock(expression, diagnostics);
```

Comment:
The new code does not have logic for skipping the `ref` in `ref` expressions, unlike [`BindLambdaExpressionAsBlock`](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Binder/Binder_Statements.cs,a1c5031bc7514949) (copied below). Is that intentional? I'm surprised that ref tests weren't affected.

```C#

        /// <summary>
        /// Binds a lambda with expression e as either { return e;} or { e; }.
        /// </summary>
        public BoundBlock BindLambdaExpressionAsBlock(ExpressionSyntax body, DiagnosticBag diagnostics)
        {
            Binder bodyBinder = this.GetBinder(body);
            Debug.Assert(bodyBinder != null);

            RefKind refKind;
            var expressionSyntax = ((ExpressionSyntax)body).SkipRef(out refKind);
            BoundExpression expression = bodyBinder.BindValue(expressionSyntax, diagnostics, refKind != RefKind.None ? BindValueKind.RefReturn : BindValueKind.RValue);
            return bodyBinder.CreateBlockFromExpression(body, bodyBinder.GetDeclaredLocalsForScope(body), refKind, expression, expressionSyntax, diagnostics);
        }
```